### PR TITLE
Feat(front) better UX about instance

### LINKED ISSUE #28: feat(front): better UX about instance
On hover display the remove button on top right. (see the screenshot below)

image

The selected instance must be visible in the list (see the screenshot below)

image

### PR DESCRIPTION
**Pull request by tomyo002**

### CODE REVIEW

**Comment by darkweak** [comment]
Put a screenshot about the result.

**Comment by tomyo002** [comment]
Capture d’écran du 2024-06-06 14-06-32

File: frontend/src/components/atomic/Icon.tsx [hunk 1]
```diff
@@ -79,5 +83,5 @@ export const Icon: React.FC<IconProps> = ({ className, name }) => {
   if (!Icon) {
     return null;
   }
-  return <Icon className={`h-6 w-6 ${className}`} />;
+  return <Icon className={!className ? 'h-6 w-6' : className} />;
```
Comment (darkweak) [inline frontend/src/components/atomic/Icon.tsx]:
```suggestion
  return <Icon className={className || 'h-6 w-6'} />;
```

Comment (darkweak) [reply frontend/src/components/atomic/Icon.tsx]: `className ?? 'h-6 w-6'`

**Review by darkweak — CHANGES_REQUESTED** [review]

**Comment by tomyo002** [comment]
Capture d’écran du 2024-06-06 15-48-39

**Comment by darkweak** [comment]
There are still differences between the expectation and your proposition

**Comment by darkweak** [comment]
@tomyo002 what about this PR?

**Comment by tomyo002** [comment]
Capture d’écran du 2024-06-11 16-22-42

**Comment by darkweak** [comment]
@tomyo002 Still not the same as expected!

**Comment by tomyo002** [comment]
Capture d’écran du 2024-06-12 16-23-40

File: frontend/src/components/layout/Menu.tsx [hunk 2]
```diff
@@ -1,45 +1,68 @@
...
+                            inst.name !== instance.name ||
+                            inst.baseUrl !== instance.baseUrl,
+                        ),
+                      );
+                      navigate(path.HOME);
+                    }}
+                  >
+                    <Icon className="text-white h-4 w-4" name="cross" />
+                  </Button>
+                )}
```
Comment (darkweak) [inline frontend/src/components/layout/Menu.tsx]: You don't need javascript to display that.

File: frontend/src/components/layout/Menu.tsx [hunk 3]
```diff
@@ -1,45 +1,68 @@
...
+  const handleMouseEnter = (instance: InstanceType) => {
+    clearTimeout(hideTimeout);
+    setHoveredInstance(instance);
+  };
+
+  const handleMouseLeave = () => {
+    hideTimeout = setTimeout(() => {
+      setHoveredInstance(undefined);
+    }, 300);
+  };
```
Comment (darkweak) [inline frontend/src/components/layout/Menu.tsx]: Basically, using `mouseEnter` and `mouseLeave` is a bad idea.

**Review by darkweak — CHANGES_REQUESTED** [review]

**Review by darkweak — APPROVED** [review]

**Pull request merged**